platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -1986,12 +1986,23 @@ bool DisplayServerX11::_window_fullscreen_check(WindowID p_window) const {
 void DisplayServerX11::_validate_mode_on_map(WindowID p_window) {
 	// Check if we applied any window modes that didn't take effect while unmapped
 	const WindowData &wd = windows[p_window];
-	if (wd.fullscreen && !_window_fullscreen_check(p_window)) {
-		_set_wm_fullscreen(p_window, true, wd.exclusive_fullscreen);
-	} else if (wd.maximized && !_window_maximize_check(p_window, "_NET_WM_STATE")) {
-		_set_wm_maximized(p_window, true);
-	} else if (wd.minimized && !_window_minimize_check(p_window)) {
-		_set_wm_minimized(p_window, true);
+
+	bool desired_fullscreen = _window_fullscreen_check(p_window);
```

Comment:
Why is it called `desired_fullscreen`? `_window_fullscreen_check` is checking if it's current full screen and desired mode is `wd.fullscreen`.

The code seems to be wrong, it will never change mode.

Reply:
I've tested both on KDE and BSPWM, all modes are working.
The issue is that  BSPWM sends MapNotify before ConfigureNotify, causing wd.minimized to be outdated. I've provided an explanation in the linked issue.

Reply:
Do we need to validate mode only on window creation? If so, would something like this work?
```cpp
WindowData &wd = windows[p_window];

if (wd.validated) return;

if (wd.fullscreen && !_window_fullscreen_check(p_window)) {
	_set_wm_fullscreen(p_window, true, wd.exclusive_fullscreen);
} else if (wd.maximized && !_window_maximize_check(p_window, "_NET_WM_STATE")) {
	_set_wm_maximized(p_window, true);
} else if (wd.minimized && !_window_minimize_check(p_window)) {
	_set_wm_minimized(p_window, true);
}

wd.validated = true;
```

Reply:
Or maybe we can just listen for PropertyNotify events involving WM_STATE changes to set them instantly?